base/linalg/blas.jl

```diff
@@ -798,8 +857,10 @@ end
     trsv!(ul, tA, dA, A, b)
 
 Overwrite `b` with the solution to `A*x = b` or one of the other two variants determined by
-`tA` (transpose `A`) and `ul` (triangle of `A` used). `dA` indicates if `A` is
-unit-triangular (the diagonal is assumed to be all ones). Returns the updated `b`.
+`tA` (transpose `A` if `'T'`, `A` if `'N'`, conjugate-transpose if `'C'`)
+and `ul` (triangle of `A` used, `'U'` for upper, `'L`' for lower).
+`dA` indicates if `A` is unit-triangular (the diagonal is assumed to be all ones if `dA` is
+`'U'`, and assumed to have non-unit values if `dA` is `'N'`). Returns the updated `b`.
```

Comment:
no, there's no assumption about the diagonal values at all in this case - they get read from